finality-provider/cmd/fpd/main.go

```diff
@@ -4,23 +4,64 @@ import (
 	"fmt"
 	"os"
 
-	dcli "github.com/babylonchain/finality-provider/finality-provider/cmd/fpd/daemon"
-	"github.com/urfave/cli"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/config"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	"github.com/spf13/cobra"
+
+	appparams "github.com/babylonchain/babylon/app/params"
+	"github.com/babylonchain/finality-provider/finality-provider/cmd/fpd/daemon"
+	fpcfg "github.com/babylonchain/finality-provider/finality-provider/config"
 )
 
-func fatal(err error) {
-	fmt.Fprintf(os.Stderr, "[btc-finality-provider] %v\n", err)
-	os.Exit(1)
+// NewRootCmd creates a new root command for fpd. It is called once in the main function.
+func NewRootCmd() *cobra.Command {
+	var (
+		clientCtx client.Context
+	)
+	encCfg := appparams.DefaultEncodingConfig()
+
+	rootCmd := &cobra.Command{
+		Use:           "fpd",
+		Short:         "fpd - Finality Provider Daemon (fpd).",
+		Long:          `fpd is the daemon to handle finality provider submission of randomness from BTC to proof of stake chains`,
```

Comment:
```suggestion
		Long:          `fpd is the daemon to create and manage finality providers`,
```
We can make it general because it also handles submission of finality signatures and serving PoS chains beyond (e.g., L2)